### DISCUSSION TITLE
Node@12 on M1 has error when using long.js

### DISCUSSION THREAD

**Discussion by jspark-gigworks** [Homebrew/discussions#640] [Everyday usage]
I have installed node@12 (v12.20.1) on M1. It has been compiled successfully.
But, it crashes when the application which includes [long.js](https://github.com/dcodeIO/long.js) is executed.

```
<--- Last few GCs --->

[95980:0x158008000]      173 ms: Scavenge 9.9 (15.5) -> 7.2 (15.7) MB, 0.4 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure
[95980:0x158008000]      211 ms: Scavenge 11.0 (16.0) -> 8.9 (17.2) MB, 0.5 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure
[95980:0x158008000]      245 ms: Scavenge 12.9 (18.0) -> 11.0 (19.7) MB, 0.5 / 0.0 ms  (average mu = 1.000, current mu = 1.000) allocation failure


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x00010fd008d1 <JSObject>
    0: builtin exit frame: new constructor(aka Module)(this=0x0001101d97c1 <Object map = 0x110c533f9>,0x0001101d9671 <Uint8Array map = 0x10fcc2b99>,0x0001101d97c1 <Object map = 0x110c533f9>)

    1: ConstructFrame [pc: 0x105808778]
    2: StubFrame [pc: 0x1058f1224]
    3: /* anonymous */ [0x1101d8f01] [/Users/anselmo/work/thalia/node_modules/@xtuc/long/src/long.js:9] [bytecode=0x110bff911 offse...
```

Long.js is very popular library, which is downloaded 8 million times weekly.

If someone wants to reproduce it, git clone [the long.js repository](https://github.com/dcodeIO/long.js) and `npm install; npm test`.

The test of long.js works well on v15.0.7 arm64 and on v12.20.0 x64.

**Comment by SMillerDev** [comment]
You should really report this to nodejs. The fact that you had to compile node@12 means that we can't get it to work sufficiently on ARM so all bug reports should be done upstream.

**Reply by jspark-gigworks** [reply]
You are right. It is caused by node itself not by homebrew.
But I think that homebrew does not link to install node@12 arm64 version if it has some issues.
I just hope to inform it to homebrew managers.

**Reply by SMillerDev** [reply]
You can try and install anything through homebrew. There's just a chance it doesn't work on M1. The only solution to this is reporting this upstream, there is nothing homebrew can do about this.
